cmd/server/main.go

```diff
@@ -39,80 +38,64 @@ type Arguments struct {
 
 func mainWithArgs(ctx context.Context, args []string, logger golog.Logger) error {
 	var argsParsed Arguments
+
 	if err := utils.ParseFlags(args, &argsParsed); err != nil {
 		return err
 	}
+
 	if argsParsed.Port == 0 {
+		golog.Global.Debugf("using default port %d ", defaultPort)
 		argsParsed.Port = utils.NetPortFlag(defaultPort)
 	}
 
-	lidarComponents := search.Devices()
-	if len(lidarComponents) != 0 {
-		golog.Global.Debugf("detected %d lidar devices", len(lidarComponents))
-		for _, comp := range lidarComponents {
+	// Check if USB device is available
+	// TODO(jeremy): add search filter for product and model info to confirm rplidar is present instead of assuming
+	usbDevices := usb.Search(
+		usb.SearchFilter{},
+		func(vendorID, productID int) bool {
+			return true
+		})
+
+	if len(usbDevices) != 0 {
+		golog.Global.Debugf("detected %d lidar devices", len(usbDevices))
+		for _, comp := range usbDevices {
 			golog.Global.Debug(comp)
 		}
-	}
-	if argsParsed.DevicePath != "" {
-		lidarComponents = []config.Component{{Type: config.ComponentTypeLidar, Model: rplidar.ModelName, Host: argsParsed.DevicePath}}
+	} else {
+		return errors.New("no usb devices found")
 	}
 
-	if len(lidarComponents) == 0 {
-		return errors.New("no lidar devices found")
+	// Create rplidar component
+	lidarDevice := config.Component{
+		Name:       "rplidar",
+		Type:       config.ComponentTypeCamera,
+		Model:      rplidar.ModelName,
+		Attributes: config.AttributeMap{"device_path": "/dev/ttyUSB0"},
```

Comment:
Why is the device path hard coded here, when above there's a USB search above for the right vendor and product id? Does the result from that not work for some reason (I know there were some issues on Mac, but...)

Reply:
Fixed it to use the search result! Thanks for pointing it out.

Done.